lms/djangoapps/certificates/queue.py

```diff
@@ -344,31 +347,32 @@ def add_cert(self, student, course_id, course=None, forced_grade=None, template_
                     cert.status = new_status
                     cert.save()
 
-                    try:
-                        self._send_to_xqueue(contents, key)
-                    except XQueueAddToQueueError as exc:
-                        new_status = ExampleCertificate.STATUS_ERROR
-                        cert.status = new_status
-                        cert.error_reason = unicode(exc)
-                        cert.save()
-                        LOGGER.critical(
-                            (
-                                u"Could not add certificate task to XQueue.  "
-                                u"The course was '%s' and the student was '%s'."
-                                u"The certificate task status has been marked as 'error' "
-                                u"and can be re-submitted with a management command."
-                            ), student.id, course_id
-                        )
-                    else:
-                        LOGGER.info(
-                            (
-                                u"The certificate status has been set to '%s'.  "
-                                u"Sent a certificate grading task to the XQueue "
-                                u"with the key '%s'. "
-                            ),
-                            key,
-                            new_status
-                        )
+                    if generate_pdf:
+                        try:
+                            self._send_to_xqueue(contents, key)
+                        except XQueueAddToQueueError as exc:
+                            new_status = ExampleCertificate.STATUS_ERROR
+                            cert.status = new_status
+                            cert.error_reason = unicode(exc)
+                            cert.save()
+                            LOGGER.critical(
+                                (
+                                    u"Could not add certificate task to XQueue.  "
+                                    u"The course was '%s' and the student was '%s'."
+                                    u"The certificate task status has been marked as 'error' "
+                                    u"and can be re-submitted with a management command."
+                                ), student.id, course_id
```

Comment:
While you're in here, could you please fix the ordering of `student.id` and `course_id`?